src/main/java/mate/academy/internetshop/controllers/product/DeleteProductController.java

```diff
@@ -0,0 +1,41 @@
+package mate.academy.internetshop.controllers.product;
+
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internetshop.lib.Injector;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.service.ProductService;
+import mate.academy.internetshop.service.ShoppingCartService;
+
+@WebServlet("/productDelete")
+public class DeleteProductController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector INJECTOR = Injector.getInstance("mate.academy.internetshop");
+    private ShoppingCartService shoppingCartService
+            = (ShoppingCartService)INJECTOR.getInstance(ShoppingCartService.class);
+    private ProductService productService
+            = (ProductService)INJECTOR.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String id = req.getParameter("productId");
+        // ***   comment to mentor   ***
+        // this controller I use to show All products for admin
+        // without creating another controller
+        // that`s why i need to check for null...
+        if (id != null) {
+            productService.delete(Long.parseLong(id));
+        }
+        List<Product> productList = productService.getAll();
+        req.setAttribute("products", productList);
+        req.getRequestDispatcher("/WEB-INF/jsp/productListAdmin.jsp").forward(req, resp);
```

Comment:
you should have two jsps one onlu with products and possibility to add them to shoppingCart and another one with possibility to delete and add new products